thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DefaultDataProvider.java

```diff
@@ -84,27 +87,38 @@ public DefaultDataProvider(MetricConfigManager metricDAO, DatasetConfigManager d
     this.timeseriesLoader = timeseriesLoader;
     this.aggregationLoader = aggregationLoader;
     this.loader = loader;
-    this.timeseriesCache = CacheBuilder.newBuilder()
-        .maximumSize(100)
-        .expireAfterWrite(30, TimeUnit.MINUTES)
-        .build(new CacheLoader<MetricSlice, DataFrame>() {
-          @Override
-          public DataFrame load(MetricSlice slice) {
-            return fetchTimeseries(Collections.singleton(slice)).get(slice);
-          }
-        });
-  }
 
+    if (DETECTION_TIME_SERIES_CACHE == null) {
+      LOG.info("initializing detection timeseries cache");
+      DETECTION_TIME_SERIES_CACHE = CacheBuilder.newBuilder()
+          // don't use more than one third of memory for detection time series
+          .maximumWeight(Runtime.getRuntime().freeMemory() / 3)
+          .expireAfterWrite(15, TimeUnit.MINUTES)
```

Comment:
Can you include a comment stating that these values are chosen randomly and needs to be tuned.

Reply:
added